cartographer_grpc/map_builder_server.cc

```diff
@@ -127,4 +139,36 @@ void MapBuilderServer::StartSlamThread() {
       [this]() { this->ProcessSensorDataQueue(); });
 }
 
+void MapBuilderServer::OnLocalSlamResult(
+    int trajectory_id, cartographer::common::Time time,
+    cartographer::transform::Rigid3d local_pose,
+    cartographer::sensor::RangeData range_data,
+    std::unique_ptr<const cartographer::mapping::NodeId> node_id) {
+  std::shared_ptr<cartographer::sensor::RangeData> shared_range_data =
+      std::make_shared<cartographer::sensor::RangeData>(std::move(range_data));
+  cartographer::common::MutexLocker locker(&local_slam_subscriptions_lock_);
+  for (auto& entry : local_slam_subscriptions_[trajectory_id]) {
+    entry.second(trajectory_id, time, local_pose, shared_range_data,
```

Comment:
Giving entry.second a name (assigning to a ref) would make this more readable

Reply:
Done.